src/assignments/tech-jobs-oo.rst

```diff
@@ -0,0 +1,580 @@
+#. Use the generator in Visual Studio to automate routine tasks.
+#. Use unit testing and Test-Driven-Development (TDD) to verify and create new
+   methods.
+#. Apply the concept of inheritance to streamline your classes (the DRY
+   idea---Don't Repeat Yourself).
+
+Get the Starter Code
+---------------------
+
+#. Set up a local copy of the project. Visit the `repository page <https://github.com/LaunchCodeEducation/csharp-web-dev-techjobs-oo>`_ 
+   for this project and fork the repo to create a copy in your own GitHub
+   account.
+#. Open Visual Studio.
+#. If the app opens up to an existing project, close it.
+#. From the Visual Studio dialog box, open your copy of the starter code.
+
+TechJobs (Object-Oriented Edition)
+----------------------------------
+
+Sally has gotten the ball rolling by adding a ``Job`` class, along with classes
+to represent the individual properties of a job: ``Employer``, ``Location``,
+``PositionType``, and ``CoreCompetency``. She completed the ``Employer`` class,
+and she left you the task of filling in the others.
+
+As the team gets closer to deploying the app---and abandoning the test data
+they’ve been using---they’ll want an easy way to add and remove jobs via a
+user interface. Before that, however, you need to finish shifting the project
+to an object-oriented design.
+
+Why Change to Object-Oriented?
+------------------------------
+
+Working with data stored as strings in dictionaries and lists isn’t a good
+long-term solution, for reasons that we point out below.
+
+The ``Job`` class introduces an object-oriented design to the application. It
+contains all of the fields you used in the console version of TechJobs:
+``Name``, ``Employer``, ``EmployerLocation``, ``JobType``, ``JobCoreCompetency``.
+There’s also an ``Id`` field which will be used to uniquely identify ``Job``
+objects.
+
+The main difference between the object representation of a job and the
+string-based representation is that the values of ``EmployerName``, ``EmployerLocation``,
+and the other non-ID fields are no longer strings. Instead, they are classes of
+their own.
+
+Job Fields
+^^^^^^^^^^^
+
+Open the ``Job`` class file. You’ll see the following fields and properties (among others):
+
+.. sourcecode:: csharp
+   :linenos:
+
+   public string Name { get; set; }
+   public Employer EmployerName { get; set; }
+   public Location EmployerLocation { get; set; }
+   public PositionType JobType { get; set; }
+   public CoreCompetency JobCoreCompetency { get; set; }
+
+Of these, only ``Name`` is a string. Sally created classes to represent each of
+the other properties. These classes---``Employer``, ``Location``,
+``CoreCompetency``, ``PositionType``---have ``Value`` and ``Id`` fields.
+
+So, for example, if you had a ``Job`` instance, you could get the name of the
+employer this way:
+
+.. sourcecode:: csharp
+
+   // job is an instance of Job
+   string employerName = job.Employer.Value;
```

Comment:
Fixed the ``Employer``! That is supposed to be ``EmployerName``!

I do like the organization as is because it gives them something to think about when it comes to streamlining the organization at the end. I did add a little note to the end of the next section to indicate that the app could be better organized and that students will need to do some work to make it so!